cluster-provision/images/vm-image-builder/create-containerdisk.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+set -exuo pipefail
+
+SCRIPT_PATH=$(dirname "$(realpath "$0")")
+
+export CUSTOMIZE_IMAGE_SCRIPT=${CUSTOMIZE_IMAGE_SCRIPT:-"${SCRIPT_PATH}/customize-image.sh"}
+
+function customize_image() {
+  local source_image=$1
+  local os_variant=$2
+  local customized_image=$3
+  local cloud_config=$4
+
+  # Backup the VM image and pass copy of the original image
+  # in case customizing script fail.
+  vm_image_copy="$(dirname $customized_image)/copy-${source_image}"
+  cp "$source_image" "$vm_image_copy"
+
+  # TODO: convert this script and its dependencies to container
+  ${CUSTOMIZE_IMAGE_SCRIPT} "$vm_image_copy" "$os_variant" "$customized_image" "$cloud_config"
+
+  # Backup no longer needed.
+  rm -f "$vm_image_copy"
+}
+
+function cleanup() {
+    if [ $? -ne 0 ];then
+      rm -rf "${build_directory}"
+    fi
+
+    rm -f "copy-${VM_IMAGE}"
+}
+
+export IMAGE_NAME=${IMAGE_NAME:-example-fedora}
+export TAG=${TAG:-32}
+export OS_VARIANT=${OS_VARIANT:-fedora31}
+export CLOUD_CONFIG_PATH=${CLOUD_CONFIG_PATH:-"${SCRIPT_PATH}/example/cloud-config"}
+export VM_IMAGE_URL=${VM_IMAGE_URL:-$(cat "${SCRIPT_PATH}/example/image-url")}
+
+readonly VM_IMAGE="source-image.qcow2"
+readonly build_directory="${IMAGE_NAME}_build"
+readonly new_vm_image_name="provisioned-image.qcow2"
+
+trap 'cleanup' EXIT SIGINT
+
+pushd "${SCRIPT_PATH}"
+  cleanup
+
+  if ! [ -e "$VM_IMAGE" ]; then
+    # Download base VM image
+    curl -L $VM_IMAGE_URL -o $VM_IMAGE
+  fi
+
+  mkdir "${build_directory}"
```

Comment:
-p , in case the previous run was fine

Reply:
I did not added `-p` so if a user run it again it wont delete the qcow2 image that was already created
What you think?

Reply:
but wont it recreate it on the same name ?
i would then fail fast if the folder exists, if the case is what is above, or just `rm -rf` the folder if the above answer is yes

Reply:
It is like you described, if you already created an image next time you run this script you will fail with an error saying directory already exists

Reply:
Sorry for the noise, at first the idea was to keep files if something went wrong but it had no use
so i changed to delete file that were created in each phase is there is an error

Reply:
i saw before that it clean just if error occour, but what if no error ?
i think that even if you choose to fail the test , i would prefer please something like
[ -d dir ] && echo "dir exist" && exit 1
(dont remember the exact check in the first one)

Reply:
If no error occurs and the `...build` directory exists it will report that the directory exists and delete it, then on the second run it will run normally